File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        
        List<List<Integer>> ll= new ArrayList<>();
        Queue<TreeNode> q= new LinkedList<>();
        if(root==null){
            return ll;
        }
        q.add(root);
        while(!q.isEmpty()){
            int size=q.size();
            List<Integer> inner= new ArrayList<>();
            for(int i=0;i<size;i++){
                TreeNode  rm= q.poll();
                inner.add(rm.val);

                if(rm.left!=null){
                    q.add(rm.left);
                }
                if(rm.right!=null){
                    q.add(rm.right);
                }
            }
            ll.add(new ArrayList<>(inner));
        }
        return ll;
    }
}

File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        
        std:: queue<TreeNode*> q;
        vector<vector<int>> ll;
        if(root==nullptr){
            return ll;
        }
        q.push(root);
        while(!q.empty()){
            int size = q.size();
            vector<int> inner;
            for(int i=0;i<size;i++){
                TreeNode* rm = q.front();
                q.pop();
                inner.push_back(rm->val);
                if(rm->left!=nullptr){
                    q.push(rm->left);
                }
                if(rm->right!=nullptr){
                    q.push(rm->right);
                }

            }
            ll.push_back(inner);
        }
        return ll;
    }
};